optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -319,7 +304,13 @@ private void processFactEntityOrScoreAnnotation(DescriptorPolicy descriptorPolic
                 || annotationClass.equals(PlanningEntityCollectionProperty.class)) {
             processPlanningEntityPropertyAnnotation(descriptorPolicy, member, annotationClass);
         } else if (annotationClass.equals(PlanningScore.class)) {
-            processScoreAnnotation(descriptorPolicy, member, annotationClass);
+            if (scoreDescriptor == null) {
+                // Bottom class wins. Bottom classes are parsed first due to ConfigUtil.getAllAnnotatedLineageClasses().
+                scoreDescriptor = ScoreDescriptor.buildScoreDescriptor(descriptorPolicy, member, solutionClass);
+            } else {
+                // If an ancestor class has a score member, it must have the same type and name.
```

Comment:
Comment softly implies that there is not other case that invalidates, but there is one: 1 solution has 2 score members

Maybe something like this:
```suggestion
                // If an ancestor class has a score member (or a solution has 2 score members), it must have the same type and name.
```

Reply:
@ge0ffrey You have a good point but the suggested change doesn't make sense. I've removed the comment and renamed the method to `failFastOnDuplicateMember()` to better reflect the purpose. Exactly which other score members are allowed is not important on this level.